kips/kip-1.md

```diff
@@ -186,33 +186,33 @@ contract UserPot is RBACWithAdmin, UserPotInterface, ERC165, UpgradeableInterfac
     require(bal >= _deposit, 'you need to pay more to register for event');
     _updateUserData(_user, msg.sender, bal - _deposit);      
   }
+
+  function withdraw() external {
+    uint256 bal = calculatePayout(msg.sender);
+    msg.sender.transfer(bal);
+    _updateUserData(msg.sender, address(0), 0);
+  }
   
-    function withdraw() external {
-        uint256 bal = calculatePayout(msg.sender);
-        msg.sender.transfer(bal);
-        _updateUserData(msg.sender, address(0), 0);
+  function _updateUserData(address _user, address _newEvent, uint256 _newBalance) internal {
+    address[] memory events = dataStore.getAddresses(_user, STORAGE_KEY_EVENTS);
+    address[] memory newEvents = new address[](10);
+    uint256 newEventsLen = 0;
+
+    for (uint256 i = 0; i < events.length; i += 1) {
+      EventInterface e = EventInterface(events[i]);
+      // remove ended events
+      if (!e.hasEnded()) {
+        newEvents[newEventsLen] = events[i];
+        newEventsLen++;
+      }
     }
-    
-    function _updateUserData(address _user, address _newEvent, uint256 _newBalance) internal {
-        address[] memory events = dataStore.getAddresses(_user, STORAGE_KEY_EVENTS);
-        address[] memory newEvents = new address[](10);
-        uint256 newEventsLen = 0;
-
-        for (uint256 i = 0; i < events.length; i += 1) {
-            EventInterface e = EventInterface(events[i]);
-            // remove ended events
-            if (!e.hasEnded()) {
-                newEvents[newEventsLen] = events[i];
-                newEventsLen++;
```

Comment:
I don't understand this logic fully. If you have registered 3 events so far (none of them ended yet) and register the forth one, you increment by 3 here and 1 at l221?